python-client/docs/reference/notebooks/insurance_prediction_sklearn.ipynb

```diff
@@ -0,0 +1,545 @@
+   ],
+   "metadata": {
+    "collapsed": false,
+    "ExecuteTime": {
+     "end_time": "2023-08-01T15:34:28.438516Z",
+     "start_time": "2023-08-01T15:34:28.429169Z"
+    }
+   },
+   "id": "2a5351b1c97f2a31"
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### Wrap dataset with Giskard"
+   ],
+   "metadata": {
+    "collapsed": false
+   },
+   "id": "121478a9aa05b571"
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 29,
+   "outputs": [],
+   "source": [
+    "raw_data = pd.concat([X_test, y_test], axis=1)\n",
+    "wrapped_data = Dataset(raw_data, target=\"charges\", name=\"insurance dataset\")"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "ExecuteTime": {
+     "end_time": "2023-08-01T15:34:29.641188Z",
+     "start_time": "2023-08-01T15:34:29.626728Z"
+    }
+   },
+   "id": "4d95d45185280742"
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "## Model training"
+   ],
+   "metadata": {
+    "collapsed": false
+   },
+   "id": "81b1ef1f34961763"
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### Define preprocessing pipeline"
+   ],
+   "metadata": {
+    "collapsed": false
+   },
+   "id": "8a0bf6e9a7a42831"
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 30,
+   "outputs": [],
+   "source": [
+    "preprocessor = ColumnTransformer(transformers=[\n",
+    "    (\"scaler\", StandardScaler(), CATEGORICAL_COLS),\n",
+    "    (\"one_hot_encoder\", OneHotEncoder(handle_unknown=\"ignore\", sparse=False), NUMERICAL_COLS),\n",
+    "])"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "ExecuteTime": {
+     "end_time": "2023-08-01T15:34:33.185135Z",
+     "start_time": "2023-08-01T15:34:33.165967Z"
+    }
+   },
+   "id": "64d42c05c8107e59"
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### Build estimator"
+   ],
+   "metadata": {
+    "collapsed": false
+   },
+   "id": "495a8040bc6735f3"
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e6332de6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pipeline = Pipeline(steps=[\n",
+    "    (\"preprocessor\", preprocessor),\n",
+    "    (\"regressor\", LGBMRegressor(n_estimators=30))\n",
+    "])\n",
+    "\n",
+    "pipeline.fit(X_train, y_train)\n",
+    "\n",
+    "y_train_pred = pipeline.predict(X_train)\n",
+    "y_test_pred = pipeline.predict(X_test)\n",
+    "\n",
+    "train_r2 = r2_score(y_train, y_train_pred)\n",
+    "test_r2 = r2_score(y_test, y_test_pred)\n",
+    "\n",
+    "print(f\"Train R2-score: {train_r2:.2f}\")\n",
+    "print(f\"Test R2-score: {test_r2:.2f}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "078e028d",
+   "metadata": {},
+   "source": [
+    "### Wrap model with Giskard"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1e13357c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "wrapped_model = Model(pipeline,\n",
```

Comment:
This will work, but since we display the `prediction_function` option in our doc, I prefer that the user passes `pipeline.predict` as a model for now